week3/xor/jasonkuhrt/Main.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE OverloadedStrings #-}
+{- README
+
+-- XOR Encryption
+
+-- Spec
+
+Write a program that encrypts a file by XORing it with a key. For instance:
+
+     Key  "XYZ"
+Contents  "abracadabra"
+     XOR   XYZXYZXYZXY
+       =   9;(9:;<88*8
+
+* It should accept the file name and the key as command-line arguments
+* Use bytestring to read the file
+* Encode the key as UTF8 (with encodeUtf8)
+* Overwrite the file with encryped variant
+
+-- Notes
+
+Due to the way XOR works, encrypting an already encrypted file will decrypt
+it. Learn more about XOR Ciphers at en.wikipedia.org/wiki/xor_cipher.
+-}
+
+
+
+module Main where
+
+import qualified Data.Bits as Bits
+import qualified System.Environment as Env
+import Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy as B
+import qualified Data.ByteString.Lazy.Char8 as BC8
+import Data.Text.Lazy (Text)
+import qualified Data.Text.Lazy as Text
+import qualified Data.Text.Lazy.Encoding as Enc
```

Comment:
Wow I had no idea one could do that! Not sure what to think about this feature though. Seems somewhat ad-hoc. It certainly doesn't work with say:

```
Preprocessing executable 'xor' for xor-0.0.0.1...
[1 of 1] Compiling Main             ( Main.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.0.0/build/xor/xor-tmp/Main.o )

/Users/jasonkuhrt/projects/haskell-ex/week3/xor/jasonkuhrt/Main.hs:75:15: error:
    Ambiguous occurrence ‘BSL.readFile’
    It could refer to either ‘BSL.readFile’,
                             imported qualified from ‘Data.ByteString.Lazy.Char8’ at Main.hs:34:1-50
                          or ‘BSL.readFile’,
                             imported qualified from ‘Data.ByteString.Lazy’ at Main.hs:33:1-44

...
```

But none the less good to know!